integration-tests/tests/src/tests/credentials/anonymous.ts

```diff
@@ -1,18 +1,14 @@
 import { expect } from "chai";
-import { App, Credentials, User } from "realm";
+import { Credentials, User } from "realm";
 
-import { importApp } from "../../utils/import-app";
+import { importAppBefore } from "../../hooks";
 
 describe.skipIf(environment.missingServer, "anonymous credentials", () => {
-  let app: App;
-  
-  before(async function() {
-    app = await importApp("simple");
-  });
+  importAppBefore("simple");
 
-  it.skip("authenticates", async function() {
+  it("authenticates", async function(this: AppContext) {
     const credentials = Credentials.anonymous();
-    const user = await app.logIn(credentials);
+    const user = await this.app.logIn(credentials);
```

Comment:
I think that setting `this.app` implicitly in the `importAppBefore` hook is perhaps too magical.  Without looking into the implementation of said hook, I would not know that `this.app` is being set.

Reply:
Yeah, I agree this is not how most Mocha tests are written and having the `this.app` injected is not obvious where it came from. However, it (the global `Mocha.Context`) is a feature of Mocha to transfer values between hooks, suites and tests. As far as I can tell, this is the purpose of the feature.

My hope is that these hooks will be used in every sync related test and if we start relying more on them, we should also ensure proper documentation of their use for every new developer on the project to read. And as such, it won't be "magical" to the average developer contributing to the tests, because they would be used everywhere in the sync tests. 🤞